piplicenses/__init__.py

```diff
@@ -188,19 +188,29 @@ def parser(expression: str) -> set[str]:
     return parser
 
 
+def parse_licenses_list(licenses_str: str | None) -> list[str]:
+    if licenses_str is None:
+        return []
+
+    licenses = licenses_str.split(";")
+
+    # Strip items
+    licenses = [license.strip() for license in licenses]
+
+    # Remove empty string items
+    licenses = [license for license in licenses if license != ""]
```

Comment:
Similar to above - is the list comprehension better than filtering?

```suggestion
    licenses = list(filter(None, licenses))
```

Reply:
:thinking: Consider filter is more resilient here because it checks for `falsy` values including the empty string.

## TL;DR
The only advantage of list comprehension is more a matter of "more pythonic" (e.g., widely used)

<details><summary>Copilot analysis (for posterity)</summary>

## 🔍 Comparison of List Comprehension vs. filter() in Python

Both approaches you provided are used to filter out empty strings from a list of licenses. Let's analyze and compare them based on various factors.

### 📝 Code Comparison

| Feature                | List Comprehension                                      | filter()                                      |
|------------------------|--------------------------------------------------------|-----------------------------------------------|
| Syntax                 | `licenses = [license for license in licenses if license != ""]` | `licenses = list(filter(None, licenses))`    |
| Readability            | More explicit; clearly shows the filtering condition   | More concise; may be less clear to beginners  |
| Performance            | Generally faster for small to medium lists            | Slightly faster for larger lists due to C implementation |
| Type of Result         | Returns a new list directly                            | Returns a filter object, which is converted to a list |
| Flexibility            | Can easily modify the condition                        | Limited to functions or `None` checks         |
| Type Safety            | Explicitly checks for empty strings                    | Implicitly checks for falsy values (e.g., `None`, `0`, `""`) |

### ⚙️ Performance Analysis

Given that the `licenses` list is not likely to exceed **10,000 items**, both methods will perform adequately. However, the performance difference may become noticeable with larger datasets:

- **List Comprehension**: This method is generally faster for smaller lists because it is optimized for Python's list creation. It directly constructs the new list in a single pass.
  
- **filter()**: This function is implemented in C, which can provide performance benefits for larger datasets. However, for lists of around 10,000 items, the difference is minimal.

### 📊 Pros and Cons

| Approach               | Pros                                                  | Cons                                                  |
|-----------------------|-------------------------------------------------------|-------------------------------------------------------|
| List Comprehension    | - Clear and explicit syntax                          | - Slightly slower for very large lists                |
|                       | - Easy to modify filtering conditions                 |                                                       |
|                       | - More Pythonic and widely used                       |                                                       |
| filter()              | - Concise and potentially faster for large lists     | - Less readable for beginners                          |
|                       | - Can be more efficient in memory usage              | - Implicitly filters out all falsy values, not just empty strings |

### 🧩 Conclusion

Both approaches are valid for filtering out empty strings from a list of licenses. If you prioritize **readability and explicitness**, the **list comprehension** method is preferable. If you prefer **conciseness** and potentially better performance with larger datasets, then **filter()** is a good choice.

Given your assumption of a list that won't exceed **10,000 items**, either method will work effectively. If you want to ensure clarity in your code, especially for others who may read it, the list comprehension approach might be the better option.
</details>